### PR TITLE
Max/truncate and copy for s3

### PR DESCRIPTION
**Pull request by maxwellsmart84**
This adds copy s3 files and copy directory.  This also solves the 1000 s3 key issue with some recursive magic.

### CODE REVIEW

File: library/s3/index.js [hunk 1]
```diff
@@ -128,6 +132,66 @@ function s3Wrapper({ accessKeyId, secretAccessKey, region }) {
...
+        await new Promise((res) => setTimeout(res, waitInterval));
+        await copyS3File(copyParams);
+        console.log(`s3://${sourceBucket}/${key} to s3://${destinationBucket}/${newKey}`);
+      }
+    }
+  }
+
+  async function copyS3File({ sourceBucket, s3SourceFile, destinationBucket, s3DestinationFile, options={} } = {}) {
+    if (!sourceBucket || !s3SourceFile || !destinationBucket || !s3DestinationFile) {
+      throw new Error(`moveS3File() requires parameters properties: sourceBucket, s3SourceFile, s3DestinationFile, and destinationBucket`);
```
Comment (rogersgt) [inline library/s3/index.js]: Error should say `copyS3File()`

File: library/s3/index.js [hunk 2]
```diff
@@ -128,6 +132,66 @@ function s3Wrapper({ accessKeyId, secretAccessKey, region }) {
     }
   }
 
+
+  async function copyS3Directory({ sourceBucket, s3SourceDirectory, destinationBucket, s3DestinationDirectory, options={} } = {}) {
+    if (!sourceBucket || !s3SourceDirectory || !destinationBucket || (s3DestinationDirectory !== '' && !s3DestinationDirectory)) {
+      throw new Error(`moveS3Directory() requires parameters: sourceBucket, s3SourceDirectory, destinationBucket, and s3DestinationDirectory`);
```
Comment (rogersgt) [inline library/s3/index.js]: Error should say `copyS3Directory()`

**Review by rogersgt — CHANGES_REQUESTED** [review]
Just update the error messages being thrown and I think this all looks good to me. Nice job 👍

**Comment by maxwellsmart84** [comment]
going to add delete and delete versions for this.

**Pull request closed without merging**